compute_tools/src/bin/fast_import.rs

```diff
@@ -359,10 +378,32 @@ async fn run_dump_restore(
         "--verbose".to_string(),
     ];
 
+    let mut pg_dump_args = vec![
+        // always include public schema objects by default
+        // but never create the schema itself, because it already exists
+        "--schema-no-create".to_string(),
+        "public".to_string(),
+        // this makes sure any unsupported extensions are not included in the dump
+        // even if we don't specify supported extensions explicitly
+        "--extension".to_string(),
+        "plpgsql".to_string(),
+    ];
+
+    for schema in &schemas {
+        pg_dump_args.push("--schema".to_string());
+        pg_dump_args.push(schema.clone());
+    }
```

Comment:
I am a little worried though that this will introduce more unclear moments if users have custom schemas. We might want to support schema selection in cplane/console and then merge/enable this part. Can we hide this under a feature flag?

Reply:
This is exactly the reason for this PR. You can't dump all schemas at all times. Those might contain objects we don't support. 99% of users will be fine with just public schema. The new parameters cover those indeed using custom schemas.

Reply:
Looks like there are lots of imports with not just public schema: https://neonprod.grafana.net/goto/n_WexzBHR?orgId=1

What is current behaviour? Without this patch, we dump all schemas, right? I guess we need to continue doing it, so we need to patch control plane as well.

Reply:
> there are lots of imports with not just public schema:

Most of which are exactly platform-specific schemas we don't want to import

Reply:
> Without this patch, we dump all schemas, right?

We attempt to dump all. And fail import exactly for that reason.

Reply:
> I guess we need to continue doing it, so we need to patch control plane as well and only then open some UI option with default values

Ok, we can. But imports from Supabase and alike won't work.

Reply:
> If a user has an issue and needs to import other schemas, let's at least give them API way to import selected schemas

That's what this PR adds. What else is missing in the implementation?

Reply:
> Let's make this behaviour non-default and enable separately in cplane at least.

Again. We can. But what you're suggesting as a default behavior simply does not work.